### ISSUE TITLE
Redundant compact most bottom level on manual compact

### ISSUE THREAD

**Issue by rockeet** [facebook/rocksdb#8556]
### Expected behavior
When all data compacted to most bottom level, the compaction should be done.

### Actual behavior
When all data are compacted to most bottom level, the manual compaction is continue running for compact the most bottom level again.

### Steps to reproduce the behavior
Run a manual compaction on full range, we find this issue by continually dump the stat "rocksdb.sstables" and check "being_compacted" flag for each sst file.

**Comment by jay-zhuang** [comment]
Are you saying if there's only the bottom-most level, manual compaction should not do anything?
Maybe you can set `BottommostLevelCompaction` to `kSkip`:
https://github.com/facebook/rocksdb/blob/f127d459adddd4d1e381541ac066610df6905f8a/include/rocksdb/options.h#L1661-L1664

**Comment by rockeet** [comment]
> Are you saying if there's only the bottom-most level, manual compaction should not do anything?
> Maybe you can set `BottommostLevelCompaction` to `kSkip`:
> https://github.com/facebook/rocksdb/blob/f127d459adddd4d1e381541ac066610df6905f8a/include/rocksdb/options.h#L1661-L1664

No, I say manual compact does unneccessary redundant work, if there are multiple input levels, it should stop once all data are compacted to the bottom most level. The actual behavior is: it does not stop, the bottom most level will be compacted again.

**Comment by wolfkdy** [comment]
@rockeet  Perpahs it is correct bahavior, and no easy way to avoid. 
Because there may be tombstones pinned by unreleased snapshots or timestamps(a feature in development) in the bottom-most level. These tombstones should be erased when snapshots are released. So doing manual bottom-most compaction are possible to make progress.

**Comment by ajkr** [comment]
Maybe we should extend #5138 to also apply with default option (BottommostLevelCompaction::kIfHaveCompactionFilter). I can't remember a reason we didn't do that initially.
